src/types/json.h

```diff
@@ -301,6 +302,38 @@ struct JsonValue {
     return popped_values;
   }
 
+  Status ArrTrim(std::string_view path, int64_t start, int64_t stop, std::vector<std::optional<uint64_t>> &results) {
+    try {
+      jsoncons::jsonpath::json_replace(
+          value, path, [&results, start, stop](const std::string & /*path*/, jsoncons::json &val) {
+            if (val.is_array()) {
+              auto len = static_cast<int64_t>(val.size());
+              auto begin_index = start < 0 ? std::max(len + start, 0l) : start;
+              auto end_index = std::min(stop < 0 ? std::max(len + stop, 0l) : stop, len - 1);
```

Comment:
Need to explicitly cast the `0` to int64 since std::max requires both elements should be the same type.
```suggestion
              auto begin_index = start < 0 ? std::max(len + start, static_cast<int64_t>(0)) : start;
              auto end_index = std::min(stop < 0 ? std::max(len + stop, static_cast<int64_t>(0)) : stop, len - 1);
```